### PR TITLE
Adding 3.7 to Travis and Appveyor 

### PR DESCRIPTION
**Pull request by heavelock**

### CODE REVIEW

**Comment by megies** [comment]
Well, actually we should have those in obspy main repo anyway, see #2247

**Comment by heavelock** [comment]
Yup, but I wanted first to solve conflicts in my fork and to do the PR when I will solve everything. As for now I see that there is a dependency conflict. I will try to solve it tomorrow.

File: .travis.yml [hunk 1]
```diff
@@ -114,7 +118,13 @@ install:
...
         PYPROJ="pyproj"
       fi
-  - conda create -q -n test-environment
+  - |
+    if [[ "${PYTHON_VERSION}" == '3.7' ]]; then
+      FREETYPE="freetype"
+    else
+      FREETYPE="freetype<2.8"
+    fi
+  - conda create -q -n test-environment numpy scipy matplotlib basemap pyproj
```
Comment (megies) [inline .travis.yml]: `numpy scipy matplotlib basemap pyproj` should already be set in the following lines, no need to add them again on this line

**Comment by megies** [comment]
@heavelock looks good.. mind adding 3.7 on appveyor as well?

**Comment by heavelock** [comment]
Sure! 

Comment about new failures:
1. SDS tests, might be connected with this https://bugs.python.org/issue27364
2. Basemap test are connected with removing `ax._hold` in matplotlib 3.0 https://matplotlib.org/api/api_changes.html#hold-machinery

**Comment by megies** [comment]
> * SDS tests, might be connected with this https://bugs.python.org/issue27364
> 
> * Basemap test are connected with removing `ax._hold` in matplotlib 3.0 https://matplotlib.org/api/api_changes.html#hold-machinery

The matplotlib/basemap guys are already aware of the latter issue, so I guess it'll go away eventually anyways.

**Pull request merged**